### PR TITLE
[BE] Task#210596 Cohort new feature flow

### PR DESCRIPTION
**Pull request by GitVivekHub**
https://tracker.tekdi.net/issues/210596

# I have ensured that following `Pull Request Checklist` is taken care of before sending this PR

* [x] Code is formatted as per format decided
* [x] Updated acceptance criteria in tracker
* [x] Updated test cases in test-cases-tracker
* [x] Updated new API endpoint if any in common postman collection
* [x] Updated DB changes in db-tracker if any

## Summary by CodeRabbit

- **Enhancements**
  - Improved user cohort retrieval by introducing a new filter option based on cohort type.
  - Restructured logic for handling different cohort types and user roles, resulting in more tailored data retrieval.

### CODE REVIEW

File: src/src/user/user.service.ts [hunk 1]
```diff
@@ -1426,66 +1426,94 @@ export class UserService {
...
-			await this.hasuraServiceFromServices.executeRawSql(sql)
-		)?.result;
+		`;
 
+			cohort_data = (
+				await this.hasuraServiceFromServices.executeRawSql(sql)
+			)?.result;
+		}
 		if (cohort_data != undefined) {
 			return res.json({
```
Comment (manojLondhe) [inline src/src/user/user.service.ts:1518]: change response to return 200 status

File: src/src/user/user.service.ts [hunk 2]
```diff
@@ -1426,66 +1426,94 @@ export class UserService {
...
-		const cohort_data = (
-			await this.hasuraServiceFromServices.executeRawSql(sql)
-		)?.result;
+		`;
 
+			cohort_data = (
+				await this.hasuraServiceFromServices.executeRawSql(sql)
+			)?.result;
+		}
 		if (cohort_data != undefined) {
```
Comment (manojLondhe) [inline src/src/user/user.service.ts:1517]: if (cohort_data && cohort_data.length > 0) {

File: src/src/user/user.service.ts [hunk 3]
```diff
@@ -1426,66 +1426,94 @@ export class UserService {
...
-				parseInt(item?.parent_ip),
-			);
-			parent_ip_id = parentIPs.toString();
+				program_organisation_condition =
+					'pu.organisation_id = po.organisation_id';
+			}
 
-			primary_table = 'program_faciltators';
-		}
+			sql = `SELECT p.id as program_id, p.name as program_name,p.state_id,
```
Comment (manojLondhe) [inline src/src/user/user.service.ts:1498]: fix code indent

File: src/src/user/user.service.ts [hunk 4]
```diff
@@ -1426,66 +1426,94 @@ export class UserService {
...
+			if (role.includes('facilitator')) {
+				primary_table = 'program_faciltators';
+
+				program_organisation_condition =
+					'CAST(pu.parent_ip as Int) = po.organisation_id';
 			}
 
-			parent_ip_id = user?.data?.program_users?.[0]?.organisation_id;
-			primary_table = 'program_users';
+			sql = `SELECT ay.id as academic_year_id, ay.name as academic_year_name
```
Comment (manojLondhe) [inline src/src/user/user.service.ts:1464]: fix code indent

File: src/src/user/user.service.ts (hunk 2 above)
Comment (GitVivekHub) [reply src/src/user/user.service.ts]: resolved

File: src/src/user/user.service.ts (hunk 3 above)
Comment (GitVivekHub) [reply src/src/user/user.service.ts:1494]: resolved

File: src/src/user/user.service.ts (hunk 4 above)
Comment (GitVivekHub) [reply src/src/user/user.service.ts:1464]: resolved

File: src/src/user/user.service.ts (hunk 1 above)
Comment (GitVivekHub) [reply src/src/user/user.service.ts]: resolved

**Pull request merged**